datastore_mongo.go

```diff
@@ -85,6 +85,23 @@ func (db *DataStoreMongo) GetDevice(id DeviceID) (*Device, error) {
 	return &res, nil
 }
 
+func (db *DataStoreMongo) DeleteDevice(id DeviceID) error {
+	s := db.session.Copy()
+	defer s.Close()
+
+	filter := bson.M{"id": id}
```

Comment:
Very likely. 

BTW. on one hand there's a default index on `_id`. On the other, bson tag `_id` has some implicit, magical meaning. I guess we can accept the latter and just update the rest of the code to use `_id` consistently.